### ISSUE TITLE
Number of Parameters

### ISSUE THREAD

**Issue by liyuqiao-888** [lihongzhao99/RRGMambaFormer#2]
Your work is really great, but I have a question about the number of parameters that I would like to ask.The number of parameters I got here is 26M, which is different from the 16M mentioned in your paper. May I ask why this is?
The results are as follows:
{'visual_extractor': 42500160, 'encoder_decoder': 26921065, 'total': 69421225}

**Comment by lihongzhao99** [comment]
Thank you for your kind words and for your close attention to our work!

The discrepancy you found is indeed correct and is due to the different dimension configurations of the memory matrix used for different datasets.

In the code, you can see the following lines:

```python
self.fc_projection = nn.Linear(4096, 512)  # fc_feats iu_xray
# self.fc_projection = nn.Linear(2048, 512)  # fc_feats mimic
```

*   The 26M parameters you obtained are likely from running the model configured for the **IU X-Ray dataset**, which uses a **4096-dimensional** input feature.
*   The 16M parameters mentioned in the paper were reported for the model configured for the **MIMIC dataset**, which uses a **2048-dimensional** input feature.

If you are running on the **MIMIC dataset**, you can switch to the 2048-dimensional configuration to see the parameter count reduce to around 16M.  Alternatively, you could also test the 2048-dimensional configuration on the IU X-Ray dataset for comparison.

Thank you for pointing this out, it's a valuable detail that helps clarify how the model is adapted for different datasets.

**Comment by liyuqiao-888** [comment]
I reread your paper, and in your paper, the reason for changing the parameters to 16M seems to be a single-layer encoder–decoder architecture.

**Comment by lihongzhao99** [comment]
Thank you very much for your careful reading and for pointing this out.

You are absolutely right that in the paper, the parameter reduction to around 16M is mainly attributed to adopting a single-layer encoder–decoder architecture on the MIMIC dataset.

However, in our implementation, we also made an additional modification when adapting the model to the IU X-Ray dataset:
we changed the dimensions (and effectively the size) of the memory matrix, as reflected in the code snippet you mentioned. This architectural adjustment—specifically the use of 4096-dim visual features instead of 2048-dim—also leads to a noticeable increase in the total number of parameters.

Therefore, both factors contribute to the discrepancy:
    - Single-layer encoder–decoder → reduces parameters on MIMIC
    - Larger memory matrix / higher-dim visual features for IU X-Ray → increases parameters

This explains why the model shows different parameter counts across datasets.
Thank you again for raising this; it helps clarify the implementation differences more thoroughly.
